### ISSUE TITLE
Cannot use Kotlin for MountSpec

### ISSUE THREAD

**Issue by wooyukit** [facebook/litho#433]
## Version
0.19.0

## Issues and Steps to Reproduce
Use Kotlin to write the MountSpec object and generate the class successfully. However, using those components will contribute to the compile error. Finally, I use java for MountSpec class and it can be compiled successfully.

**Comment by pavlospt** [comment]
Hey @wooyukit , could you please post your compilation error and a small repro?

**Comment by wooyukit** [comment]
Let me explain more about my finding today. I am using feature module for instant app. I used Kotlin to create a spec in base module. It throws the Kotlindebugcompile error when using component in other feature module. But the component can be generated successfully. Really weird.

**Comment by wooyukit** [comment]
`/fragment/HomeFragment.kt: (39, 23):Unresolved reference: create
:home:compileDebugKotlin FAILED`

**Comment by wooyukit** [comment]
I found that the generated file with kotlin metadata which contributed to the compile failure.
import kotlin.Metadata;
/**
 * @see com.scmp.scmpapp.view.lithography.component.RailIndicatorSpec
 */
@Metadata(
    mv = {
        1,
        1,
        11
    },
    bv = {
        1,
        0,
        2
    },
    k = 1,
    d1 = "\u0000\u0018\n"
            + "\u0002\u0018\u0002\n"
            + "\u0002\u0010\u0000\n"
            + "\u0002\b\u0002\n"
            + "\u0002\u0018\u0002\n"
            + "\u0000\n"
            + "\u0002\u0018\u0002\n"
            + "\u0000\bÇ\u0002\u0018\u00002\u00020\u0001B\u0007\b\u0002¢\u0006\u0002\u0010\u0002J\u0010\u0010\u0003\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u0006H\u0007¨\u0006\u0007",
    d2 = {
        "",
        "()V",
        "onCreateLayout",
        "Lcom/facebook/litho/Component;",
        "c",
        "Lcom/facebook/litho/ComponentContext;",
        "base_debug"
    }
)

**Comment by pavlospt** [comment]
Can you please create a sample project so that we can investigate :) ?

**Comment by wooyukit** [comment]
Please find the sample project below to reproduce the issue.
https://github.com/wooyukit/litho-demo

**Comment by pavlospt** [comment]
@wooyukit Perfect, will check it out :)

**Comment by pavlospt** [comment]
@wooyukit Where is your MountSpec component located?

**Comment by wooyukit** [comment]
I found that the LayoutSpec has the same issue

**Comment by pavlospt** [comment]
There is a problem with your feature modules setup. It is not related to Litho because if you move the `HomeActivity` on `base` project it will compile just fine. Somehow the `home` feature module is not able to access generated classes from the `base` one. Check if you can configure the `sourceSets` to contain the generated classes as well.

**Comment by wooyukit** [comment]
Actually, I need to add the view component in the base module for sharing. if I changed to use java to write the spec, everything is fine.

**Comment by pavlospt** [comment]
Yes it seems that it is Kotlin related only!

**Comment by wooyukit** [comment]
From my point of view, the spec will generate java code which includes Kotlin.metadata annotation which is internal class cannot be called cross module. Could you edit the code generation logic to remove the Kotlin.Metadata in generated code to solve this issue?

**Comment by pavlospt** [comment]
The generated class is `public`. I am not sure I understand what you mean!

**Comment by wooyukit** [comment]
Right. I hope the litho can support Kotlin better in the future 👏🏻 since more and more developers will shift to use Kotlin for Android development.

**Comment by pavlospt** [comment]
Litho does support Kotlin and it has 2 modules with Kotlin examples, mirroring any Java functionality available in Java samples :)

**Comment by wooyukit** [comment]
Could you share the 2 modules with Kotlin example to me for reference.

**Comment by miroslav404** [comment]
Why is this issue closed? Issue is still present :( 
In this case writing component spec in Java is workaround not solution
